### PR TITLE
feat: add support to multistage dockerfile 1825

### LINKED ISSUE #1825: Enable the `jina hub build` to use Dockerfile with multi-stage building
**Describe the feature**
Enable the `jina hub build` to use Dockerfile with multi-stage building

**Your proposal**

---

**Environment**

**Screenshots**

### PR DESCRIPTION
**Pull request by nan-wang**
resolved #1825

- Raise `DockerLoginFailed` error instead of swallowing exceptions when login failed
- Fix a bug in `_image_version_exists`
- Unify the field name to `jina-version` in both `manifest.yml` and [https://hubapi.jina.ai/images](https://hubapi.jina.ai/images). 
- Update the sample manifest in `tests/unit/docker/test_hubapi.py`

### CODE REVIEW

**Comment by tadej-redstone** [comment]
Hi @nan-wang , I believe it would be better to pass labels using the `label` arg to docker SDK, as was attempted in #1732 , and to avoid the modification of the dockerfile.

The only issue was #1777. I believe I discovered the problem there (see my last comment), however the deeper problem is that the test didn't catch anything.

File: jina/docker/hubio.py [hunk 1]
```diff
@@ -627,12 +627,14 @@ def _get_revised_dockerfile(self, dockerfile_path: str, manifest: Dict) -> str:
         # modify dockerfile
         revised_dockerfile = []
         with open(dockerfile_path) as fp:
-            for l in fp:
+            _last_from_ln = None
+            for _ln, l in enumerate(fp):
```
Comment (Yongxuanzhang) [inline jina/docker/hubio.py:631]: Just curious about what does _ln and l stand for(like lines, labels?) here?

Comment (nan-wang) [reply jina/docker/hubio.py:631]: i used it for `line_numbers` :)

**Comment by nan-wang** [comment]
The `build()` function in `HubIO` class is quite complicated. I'll try to do refactoring on it and reduce the complexity a bit.

**Comment by nan-wang** [comment]
Due to the conflicts with the hub bookkeeping, I will withdraw this PR for the time being.

**Pull request closed without merging**